platform/src/main/java/org/stellar/anchor/platform/callback/RestRateIntegration.java

```diff
@@ -68,29 +88,186 @@ public GetRateResponse getRate(GetRateRequest request) throws AnchorException {
+          if (!isPositiveNumber(feeDescription.getAmount())) {
+            logErrorAndThrow(
+                "'fee.details[?].description.amount' is missing or not a positive number in the GET /rate response",
+                ServerErrorException.class);
+          }
+
+          if (!NumberHelper.hasProperSignificantDecimals(
+              feeDescription.getAmount(), feeAsset.getSignificantDecimals())) {
+            logErrorAndThrow(
+                "'fee.details[?].description.amount' has incorrect number of significant decimals in the GET /rate response",
+                ServerErrorException.class);
+          }
+          if (feeDescription.getName() == null) {
+            logErrorAndThrow(
+                "'fee.details.description[?].name' is missing in the GET /rate response",
+                ServerErrorException.class);
+          }
+          if (!isPositiveNumber(feeDescription.getAmount())) {
+            logErrorAndThrow(
+                "'fee.details[?].description.amount' is missing or not a positive number in the GET /rate response",
+                ServerErrorException.class);
+          }
+          totalFee = totalFee.add(new BigDecimal(feeDescription.getAmount()));
+        }
+
+        // check that sell_amount is equal to price * buy_amount + (fee ?: 0)
+        if (totalFee.compareTo(new BigDecimal(fee.getTotal())) != 0) {
+          logErrorAndThrow(
+              "'sell_amount' is not equal to price * buy_amount + (fee ?: 0) to  in the GET /rate response",
+              ServerErrorException.class);
+        }
+      }
+    } else {
+      // when fee is not present, check that sell_amount is equal to price * buy_amount
+      BigDecimal expected =
+          new BigDecimal(rate.getPrice()).multiply(new BigDecimal(rate.getBuyAmount()));
+      // Since we don't know how the anchor rounds the amounts, we need to check the equality in
+      // all allowed rounding modes
+      if (equalsInScale(
+          new BigDecimal(rate.getSellAmount()), expected, sellAsset.getSignificantDecimals())) {
+        logErrorAndThrow(
+            "'sell_amount' is not equal to price * buy_amount in the GET /rate response",
+            ServerErrorException.class);
       }
     }
-    return getRateResponse;
+  }
+
+  /**
+   * Check if two BigDecimals are equal in scale.
+   *
+   * <p>Different rounding modes are applied to test the equality. If the amounts do not equal in
+   * all allowed rounding modes, it returns false.
+   *
+   * <p>TODO: Returns true in release 2.9.0 to disable the total amount comparison. Will readd in
+   * 3.0.0
+   *
+   * @param amount The amount to be compared
+   * @param expected The expected amount
+   * @param scale The scale to be used for comparison
+   * @return true if the amounts are equal in scale with any of the allowed rounding modes, false
+   *     otherwise
+   */
+  private boolean equalsInScale(BigDecimal amount, BigDecimal expected, int scale) {
```

Comment:
Can we do abs approach for now? Otherwise we are not checking anything. It will return true for any 2 numbers, so this validation does nothing

Reply:
It is in `abs` approach.